pkg/query/flamegraph_arrow.go

```diff
@@ -1786,6 +1870,13 @@ func isLocationRoot(beg, end, i int64, list *array.List) bool {
 	return false
 }
 
+func isLocationLineLeaf(locations, lines *array.List, locationOffset, location, lineOffset, line int64) bool {
+	if locations.ListValues().IsValid(int(location)) && lines.ListValues().IsValid(int(line)) {
+		return location == locationOffset && line == lineOffset
```

Comment:
this is incorrect, it needs to be the "last valid in the list", it's very possible that the location at `locationOffset` was filtered out, which would mean that nothing will ever be the leaf with the current logic

Reply:
Actually, we have the same thing in the table and fixed it there. I'll update and use this line from the table: 
https://github.com/parca-dev/parca/blob/7ff555c749a8bac2ef9a13b64923245bc4d4964c/pkg/query/table.go#L158

Reply:
ah perfect, yes